src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/ChunkCrossOvers.java

```diff
@@ -0,0 +1,248 @@
+    withChunk(list, otherList, maxChunkSize, prng, (fromPos, toPos, chunk) -> {
+      for (int i = 0; i < chunk.size(); i++) {
+        operation.apply(toPos + i, chunk.get(i));
+      }
+    });
+  }
+
+  static <K, V> void insertChunk(
+      Map<K, V> map, Map<K, V> otherMap, int maxSize, PseudoRandom prng) {
+    int originalSize = map.size();
+    int maxChunkSize = Math.min(maxSize - originalSize, otherMap.size());
+    withChunk(map, otherMap, maxChunkSize, prng, (fromIterator, toIterator, chunkSize) -> {
+      // insertChunk only inserts new entries and does not overwrite existing
+      // ones. As skipping those entries would lead to fewer insertions than
+      // requested, loop over the rest of the map to fill the chunk.
+      while (map.size() < originalSize + chunkSize && fromIterator.hasNext()) {
+        Entry<K, V> entry = fromIterator.next();
+        if (!map.containsKey(entry.getKey())) {
+          map.put(entry.getKey(), entry.getValue());
+        }
+      }
+    });
+  }
+
+  static <K, V> void overwriteChunk(Map<K, V> map, Map<K, V> otherMap, PseudoRandom prng) {
+    int maxChunkSize = Math.min(map.size(), otherMap.size());
+    withChunk(map, otherMap, maxChunkSize, prng, (fromIterator, toIterator, chunkSize) -> {
+      // As keys can not be overwritten, only removed and new ones added, this
+      // cross over overwrites the values. Removal of keys is handled by the
+      // removeChunk mutation. Value equality is not checked here.
+      for (int i = 0; i < chunkSize; i++) {
+        Entry<K, V> from = fromIterator.next();
+        Entry<K, V> to = toIterator.next();
+        to.setValue(from.getValue());
+      }
+    });
+  }
+
+  static <K, V> void crossOverChunk(Map<K, V> map, Map<K, V> otherMap,
+      SerializingMutator<K> keyMutator, SerializingMutator<V> valueMutator, PseudoRandom prng) {
+    if (prng.choice()) {
+      crossOverChunkKeys(map, otherMap, keyMutator, prng);
+    } else {
+      crossOverChunkValues(map, otherMap, valueMutator, prng);
+    }
+  }
+
+  private static <K, V> void crossOverChunkKeys(
+      Map<K, V> map, Map<K, V> otherMap, SerializingMutator<K> keyMutator, PseudoRandom prng) {
+    int maxChunkSize = Math.min(map.size(), otherMap.size());
+    withChunk(map, otherMap, maxChunkSize, prng, (fromIterator, toIterator, chunkSize) -> {
+      // Remove keys only after all cross-overs to prevent generating them anew.
+      List<K> keysToRemove = new ArrayList<>(chunkSize);
+      Map<K, V> entriesToAdd = new HashMap<>(chunkSize);
+
+      // If a key cross over results in an already existing key or null, it is
+      // retried up to MAX_FAILED_CROSS_OVER_ATTEMPTS times.
+      for (int i = 0; i < chunkSize; i++) {
+        Entry<K, V> to = toIterator.next();
+        Entry<K, V> from = fromIterator.next();
+
+        // Keys are still in "map" and would be mutated without detaching them
+        // prior to the cross-over. Keys from "otherMap" are not mutated by
```

Comment:
As far as I understand the logic here, the reason for this being safe is that `otherMap` can be freely mutated (by the contract of `crossOver`) and every particular key is only tried once since the iterator advances. Could you double-check?

Reply:
Removed the retry and refactored the function to get rid of the detach.